### PR TITLE
[BUGFIX] Fix an undefined name created in #2666

### PR DESCRIPTION
**Pull request by cclauss**
#2666 changed `df` into `data` but [missed this instance](https://github.com/great-expectations/great_expectations/blame/develop/great_expectations/datasource/data_connector/data_connector.py#L352-L374) leading to an _undefined name_ which `if pretty_print:` would raise a NameError at runtime.

$ `flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics`
See #1940 for a description of these tests.
```
./great_expectations/datasource/data_connector/data_connector.py:372:29: F821 undefined name 'df'
        if pretty_print and df is not None:
                            ^
```

@alexsherstinsky @cdkini Your review please.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/datasource/data_connector/data_connector.py [hunk 1]
```diff
@@ -369,7 +369,7 @@ def _self_check_fetch_batch(
             )
         )
 
-        if pretty_print and df is not None:
+        if pretty_print and data is not None:
```
Comment (alexsherstinsky) [inline great_expectations/datasource/data_connector/data_connector.py:372]: Thank you so much, @cclauss -- great catch!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

File: great_expectations/datasource/data_connector/data_connector.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/datasource/data_connector/data_connector.py:372]: @cclauss If you can, please also get the `changelog-bot` to pass its checks in Azure, and then should be good for a merge.  Thank you so much!

**Review by cdkini — APPROVED** [review]
Also LGTM but please resolve @alexsherstinsky comment before merging :)

**Pull request merged**